developer/system/template-manager.md

```diff
@@ -0,0 +1,241 @@
+[#1403]: https://github.com/QubesOS/qubes-issues/issues/1403
+[#4518]: https://github.com/QubesOS/qubes-issues/issues/4518
+[#2062]: https://github.com/QubesOS/qubes-issues/issues/2062
+[#2064]: https://github.com/QubesOS/qubes-issues/issues/2064
+[#2534]: https://github.com/QubesOS/qubes-issues/issues/2534
+[#3040]: https://github.com/QubesOS/qubes-issues/issues/3040
+[discussion thread 1]: https://groups.google.com/forum/#!topic/qubes-devel/rwc2_miCNNE/discussion
+[discussion thread 2]: https://groups.google.com/forum/#!topic/qubes-users/uQEUpv4THsY/discussion
+
+## Features
+
+- Install/reinstall/downgrade/upgrade templates, either from local packages or remote repositories
+  - Ability to install templates in alternative pools
+  - Possibility for the template package to specify options such as the kernel or `virt_mode` used
+- List and show information about local and available templates
+  - Machine-readable output for easy extension
+- Search for templates
+- Remove templates
+  - Ability to disassociate or remove VMs based on the template to be removed
+- Show available repositories
+- Works in both dom0 and management VMs by utilizing the Admin API
+- Works well with existing tools
+- Command-line interface with DNF-like usage
+- A graphical interface also available
+
+## Package Format
+
+The RPM package format is still used.
+However, the contents are manually extracted instead of installing the whole package.
+This allows us to take advantage of existing tools for things like repository management.
+We can also avoid the burden of dealing with verification, reducing the risk of issues like [QSB-28].
+
+The package name should be in the form `qubes-template-${template_name}`.
+
+For the package metadata (e.g., summary, description), one needs to ensure that the character `|` does not appear.
+This is because we use the character as a separator internally.
+Failing to do so may result in `qvm-template` to fail with cryptic error messages.
+Note that as we already consider the repository metadata untrusted, this should not result in security issues.
+
+The file structure should be quite similar to previous template RPMs.
+Namely, there should be the following files in the package:
+
+- `var/lib/qubes/vm-templates/${template_name}/root.img.part.{00,01,...}`
+  - Split tarball of template `root.img`
+  - Note that the file is still split due to tools such as `rpm2cpio` not supporting large files.
+    (Notably, the cpio format does not support files over 4GiB.)
+- `var/lib/qubes/vm-templates/${template_name}/template.conf`
+  - Stores custom package metadata (as RPM does not support custom attributes).
+  - Uses `KEY=VALUE` format.
+  - Fields (corresponding to [qvm-prefs]/[qvm-features] tags with the same names)
+    - `virt_mode`
+      - Setting this to `pv` requires user confirmation.
+    - `kernel`
+      - Only allowed to be set to "", i.e., "none", for PVGrub.
+    - `no-monitor-layout`
```

Comment:
I would rephrase that, someone who is just scanning the list to find something may not notice that five lines below there's a description of those fields. Maybe something like:
- six boolean flags (permitted values: `1` and `0`, corresponding to true and false respectively): `no-monitor-layout`, `pci-e820-host`, `linux-stubdom`, `gui`, `gui-emulated`, `qrexec`